query/query.go

```diff
@@ -0,0 +1,202 @@
+			}, "unable to create query because a query with same title already exists in the space by same creator")
+			return errors.NewDataConflictError(fmt.Sprintf("query already exists with title = %s , space_id = %s, creator = %s", q.Title, q.SpaceID, q.Creator))
+		}
+		log.Error(ctx, map[string]interface{}{}, "error adding Query: %s", err.Error())
+		return err
+	}
+	return nil
+}
+
+// List all queries in a space
+func (r *GormQueryRepository) List(ctx context.Context, spaceID uuid.UUID) ([]Query, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "Query", "list"}, time.Now())
+	var objs []Query
+	err := r.db.Where("space_id = ?", spaceID).Find(&objs).Error
+	if err != nil && err != gorm.ErrRecordNotFound {
+		return nil, err
+	}
+	return objs, nil
+}
+
+// ListByCreator all queries in a space by a creator
+func (r *GormQueryRepository) ListByCreator(ctx context.Context, spaceID uuid.UUID, creatorID uuid.UUID) ([]Query, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "Query", "listbycreator"}, time.Now())
+	var objs []Query
+	err := r.db.Where("space_id = ? AND creator=?", spaceID, creatorID).Find(&objs).Error
+	if err != nil && err != gorm.ErrRecordNotFound {
+		return nil, err
+	}
+	return objs, nil
+}
+
+// Load Query in a space
+func (r *GormQueryRepository) Load(ctx context.Context, ID uuid.UUID, spaceID uuid.UUID) (*Query, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "query", "show"}, time.Now())
+	q := Query{}
+	tx := r.db.Where("id = ? and space_id = ?", ID, spaceID).First(&q)
+	if tx.RecordNotFound() {
+		log.Error(ctx, map[string]interface{}{
+			"query_id": ID.String(),
+		}, "record not found")
+		return nil, errors.NewNotFoundError("query", ID.String())
+	}
+	if tx.Error != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err":      tx.Error,
+			"query_id": ID.String(),
+		}, "unable to load the query by ID")
+		return nil, errors.NewInternalError(ctx, tx.Error)
+	}
+	return &q, nil
+}
+
+// IsDuplicate checks if there is already a query with same content by same user
+func (r *GormQueryRepository) IsDuplicate(ctx context.Context, q *Query) (bool, error) {
+	var exists bool
+	tableName := Query{}.TableName()
+	query := fmt.Sprintf(`
+		SELECT EXISTS (
+			SELECT 1 FROM %[1]s
+			WHERE
+				title = $1
+				AND creator = $2
+				AND space_id = $3
+				AND deleted_at IS NULL
+		)`, tableName)
+	err := r.db.CommonDB().QueryRow(query, q.Title, q.Creator, q.SpaceID).Scan(&exists)
+	if exists {
+		return true, errors.NewDataConflictError(fmt.Sprintf("query already exists with title = %s , space_id = %s, creator = %s", q.Title, q.SpaceID, q.Creator))
+	}
+	if err != nil {
+		return false, err
+	}
+	return exists, nil
+}
+
+// Delete deletes the query with the given id, returns NotFoundError or InternalError
+func (r *GormQueryRepository) Delete(ctx context.Context, ID uuid.UUID) error {
+	defer goa.MeasureSince([]string{"goa", "db", "query", "delete"}, time.Now())
+	if ID == uuid.Nil {
```

Comment:
Why treat `uuid.Nil` as a different value instead of just passing it on?

Reply:
true that, fixed and added a test in https://github.com/fabric8-services/fabric8-wit/pull/1824/commits/a79d9d40df2ccb905dfd34647bf91b4d0d76482a